File: src/main/java/org/cloudfoundry/credhub/util/InstantMillisecondsConverter.java
package org.cloudfoundry.credhub.util;

import java.time.Instant;

import javax.persistence.AttributeConverter;

public class InstantMillisecondsConverter implements AttributeConverter<Instant, Long> {

  @Override
  public Long convertToDatabaseColumn(final Instant attribute) {
    return attribute.toEpochMilli();
  }

  @Override
  public Instant convertToEntityAttribute(final Long dbData) {
    return Instant.ofEpochMilli(dbData);
  }
}

File: components/time-support/src/main/kotlin/org/cloudfoundry/credhub/util/InstantMillisecondsConverter.kt
package org.cloudfoundry.credhub.util

import jakarta.persistence.AttributeConverter
import java.time.Instant

class InstantMillisecondsConverter : AttributeConverter<Instant, Long> {
    override fun convertToDatabaseColumn(attribute: Instant): Long? = attribute.toEpochMilli()

    override fun convertToEntityAttribute(dbData: Long?): Instant = Instant.ofEpochMilli(dbData!!)
}
